### PR TITLE
Add ipyparallel

### PR DESCRIPTION
**Pull request by jakirkham**
Adds a recipe to build `ipyparallel`. Based roughly off the one in conda-recipes.

### CODE REVIEW

File: recipes/ipyparallel/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    - ipyparallel.serialize
+
+about:
+  home: http://ipython.org
+  license: BSD 3-Clause
+  summary: Interactive Parallel Computing with IPython
+
+extra:
+  recipe-maintainers:
+    - jakirkham
```
Comment (jakirkham) [inline recipes/ipyparallel/meta.yaml]: @minrk, would you be willing to be added as a maintainer? Is there anyone else we should ask about signing up too? Also, does this recipe look ok to you?

**Comment by jakirkham** [comment]
@ocefpaf, could you please review this when you have a chance? Should be quick.

File: recipes/ipyparallel/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  md5: e771add417b0d1b98d3ae92099becce5
+
+build:
+  number: 0
+  script: python setup.py install
+
+requirements:
+  build:
+    - python
+    - futures               # [py<32]
```
Comment (ocefpaf) [inline recipes/ipyparallel/meta.yaml]: I don't like all those spaces, but that is unimportant.

**Comment by ocefpaf** [comment]
LGTM. Merge away!

File: recipes/ipyparallel/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  version: {{ version }}
+
+source:
+  fn: ipyparallel-{{ version }}.tar.gz
+  url: https://pypi.python.org/packages/source/i/ipyparallel/ipyparallel-{{ version }}.tar.gz
+  md5: e771add417b0d1b98d3ae92099becce5
+
+build:
+  number: 0
+  script: python setup.py install
```
Comment (jakirkham) [inline recipes/ipyparallel/meta.yaml]: It doesn't use `setuptools` unless we are installing in develop mode or we are building some sort of `bdist` for shipping.

**Comment by jakirkham** [comment]
Thanks, @ocefpaf.

File: recipes/ipyparallel/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,56 @@
+{% set version = "5.0.1" %}
+
+package:
+  name: ipyparallel
+  version: {{ version }}
+
+source:
+  fn: ipyparallel-{{ version }}.tar.gz
+  url: https://pypi.python.org/packages/source/i/ipyparallel/ipyparallel-{{ version }}.tar.gz
```
Comment (pelson) [inline recipes/ipyparallel/meta.yaml]: Straight to the source? https://github.com/ipython/ipyparallel/archive/5.0.1.tar.gz

File: recipes/ipyparallel/meta.yaml (hunk 1 above)
Comment (minrk) [inline recipes/ipyparallel/meta.yaml]: yes and yes

File: recipes/ipyparallel/meta.yaml (hunk 3 above)
Comment (minrk) [inline recipes/ipyparallel/meta.yaml]: My personal habit for conda packages is to use `pip install --no-deps .`, because it ensures two things:

- packages that *do* import setuptools unconditionally don't lay an egg
- packages that *don't* import setuptools for install (specifically to avoid the egg), can still have their setuptools extras_require, etc. (IPython does this)

I have a hard time thinking of any cases where `setup.py install` is preferable to `pip install .`.

Comment (pelson) [inline recipes/ipyparallel/meta.yaml]: 3\. It is the future...?

Personal confession, I *never* ```pip install .```. That said, it might be the healthy thing to be doing on conda-forge.

File: recipes/ipyparallel/meta.yaml (hunk 1 above)
Comment (jakirkham) [inline recipes/ipyparallel/meta.yaml]: Thanks for your willingness to help out as always. While we certainly appreciate your willingness to sign up for these, I think we would like to make sure we are not overburdening you. So, if there are other Jupyter members that you think would be good candidates on recipes, please let us know and we can ask them as well.

File: recipes/ipyparallel/meta.yaml (hunk 3 above)
Comment (jakirkham) [inline recipes/ipyparallel/meta.yaml]: We have been adding other options like `--single-version-externally-managed --record record.txt` when using packages with `setuptools`, which has worked nicely so far. Though you do make a good point as `pip install --no-deps .` would avoid us having to check this at all. However, leaving `--no-deps` off could be ok as this will raise red flags if we missed adding a dependency as `conda` terminates any build if it tries to download requirements with `pip`.

File: recipes/ipyparallel/meta.yaml (hunk 4 above)
Comment (jakirkham) [inline recipes/ipyparallel/meta.yaml]: So, I'm not sure that I like that as the default. Here is my reasoning (partially motivated by @ocefpaf).

When one makes a PyPI release, it is basically impossible to overwrite the file submitted with the release. However, with GitHub tags this is trivial. Now it's true that people shouldn't do that, but the fact that it is possible and one might be tempted to is enough of a reason for me to steer clear.

The checksum sort of prevents against this, but then if it does become inaccurate we aren't sure if the release has been modified in some way, if we are having connectivity issues, or worse someone is tampering with what we are receiving. If we determine it has been modified (probably by repeated attempts), it is hard for us to tell as the old tag doesn't exist any more and we might not know the commit (assuming it even still exists).

TL;DR PyPI gives us some nice guarantees that GitHub doesn't. These guarantees protect us and our users. We should really embrace them.

File: recipes/ipyparallel/meta.yaml (hunk 1 above)
Comment (pelson) [inline recipes/ipyparallel/meta.yaml]: @jakirkham - I've been meaning to contact the jupyter mailing list along those lines (with links to each of the feedstocks). Are you interested in sending that message maybe?

Comment (jakirkham) [inline recipes/ipyparallel/meta.yaml]: So, I posted [this]( https://gitter.im/jupyter/jupyter?at=56eb4533c081db78786b1a68 ) to gitter some time ago.

Comment (jakirkham) [inline recipes/ipyparallel/meta.yaml]: While I definitely intended to add the bulk of Jupyter packages, you were too fast for me. :runner: :smile:

That's ok though. I think I have tried to explicitly state my interest on some of them. Will add myself and/or more Jupyter packages soon.

It is probably a good idea to add this to the Jupyter mailing list too. Might get better reach that way.

Comment (pelson) [inline recipes/ipyparallel/meta.yaml]: I think a similar message to the mailing list (which has threads) would be a good thing to do, if you are happy to do it? We can also be more concrete now, as we have all of the feedstocks to point to...

Comment (jakirkham) [inline recipes/ipyparallel/meta.yaml]: Absolutely, I'll try to write a draft and share it with you, @ocefpaf, and whoever else you guys think is relevant. Would be good to make this a little bit more formal.

Comment (minrk) [inline recipes/ipyparallel/meta.yaml]: Probably just me on this particular repo, but a ping to the list would be great. The chat doesn't get seen by many who aren't paying attention at the time.
